lib/AppInfo/Application.php

```diff
@@ -273,4 +275,52 @@ protected function registerChatHooks(EventDispatcherInterface $dispatcher) {
 		};
 		$dispatcher->addListener(Room::class . '::postDeleteRoom', $listener);
 	}
+
+	protected function registerRoomHooks(EventDispatcherInterface $dispatcher) {
+		$selfJoinedUserIdsPreDisconnect = [];
+		$listener = function(GenericEvent $event) use (&$selfJoinedUserIdsPreDisconnect) {
+			/** @var Room $room */
+			$room = $event->getSubject();
+
+			$users = $room->getParticipants()['users'];
```

Comment:
Oh wait, the event is missing the information who is disconnecting?!
Let's fix that instead of doing such crazy stuff in here

Reply:
Originally I tried to fix the event data, but ended with the current approach. I have no idea why... but I hope that I had a good reason for that hack :-P